### PR TITLE
Clean up: Deduplicate fork spec tests pasta

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

These tests have been copy/pasted over the last forks when they generally differ by a few lines.
This PR is an exercise to show how we can do better to reduce code with better abstractions.

**Which issues(s) does this PR fix?**

**Other notes for review**

- I removed ValidatePayloadHeaderWhenMergeCompletes, ValidatePayloadHeader, ProcessPayloadHeader.
These methods were no different than ValidatePayloadWhenMergeCompletes, ValidatePayload, and
ProcessPayload
- Moved VerifyBlobCommitmentCount to blocks package

### CODE REVIEW

File: beacon-chain/core/blocks/payload.go [hunk 1]
```diff
@@ -200,13 +202,13 @@ func ValidatePayload(st state.BeaconState, payload interfaces.ExecutionData) err
 //	    block_hash=payload.block_hash,
 //	    transactions_root=hash_tree_root(payload.transactions),
 //	)
-func ProcessPayload(st state.BeaconState, payload interfaces.ExecutionData) (state.BeaconState, error) {
-	var err error
-	if st.Version() >= version.Capella {
-		st, err = ProcessWithdrawals(st, payload)
```
Comment (james-prysm) [inline beacon-chain/core/blocks/payload.go:206]: do we not need to process withdrawals here?

File: beacon-chain/core/transition/transition_no_verify_sig.go [hunk 2]
```diff
@@ -328,20 +327,18 @@ func ProcessBlockForStateRoot(
 		if err != nil {
 			return nil, err
 		}
-		if blk.IsBlinded() {
-			state, err = b.ProcessPayloadHeader(state, executionData)
-		} else {
-			state, err = b.ProcessPayload(state, executionData)
+		if state.Version() >= version.Capella {
+			state, err = b.ProcessWithdrawals(state, executionData)
```
Comment (james-prysm) [inline beacon-chain/core/transition/transition_no_verify_sig.go:331]: oh i see it got added here

File: beacon-chain/core/blocks/payload.go [hunk 3]
```diff
@@ -220,70 +222,20 @@ func ProcessPayload(st state.BeaconState, payload interfaces.ExecutionData) (sta
...
-		return err
-	}
-	if !bytes.Equal(header.ParentHash(), h.BlockHash()) {
-		return ErrInvalidPayloadBlockHash
-	}
-	return nil
-}
-
-// ValidatePayloadHeader validates the payload header.
-func ValidatePayloadHeader(st state.BeaconState, header interfaces.ExecutionData) error {
```
Comment (james-prysm) [inline beacon-chain/core/blocks/payload.go:245]: nice so process payload does the same thing as process payload header

File: beacon-chain/core/transition/transition_no_verify_sig.go [hunk 4]
```diff
@@ -328,20 +327,18 @@ func ProcessBlockForStateRoot(
...
+			if err != nil {
+				return nil, errors.Wrap(err, "could not process withdrawals")
+			}
 		}
+		state, err = b.ProcessPayload(state, blk.Body())
```
Comment (james-prysm) [inline beacon-chain/core/transition/transition_no_verify_sig.go:336]: great we don't need to do process payload header checks anymore

**Review by james-prysm — APPROVED** [review]
nice PR a lot of cleanup, really like the payload changes

File: testing/spectest/shared/common/operations/test_runner.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	beaconState, err := operationFn(context.Background(), preState, wsb)
+	if postSSZExists {
+		require.NoError(t, err)
+		comparePostState(t, postSSZFilepath, sszToState, preState, beaconState)
+	} else {
+		// Note: This doesn't test anything worthwhile. It essentially tests
+		// that *any* error has occurred, not any specific error.
+		if err == nil {
+			t.Fatal("Did not fail when expected")
+		}
```
Comment (saolyn) [inline testing/spectest/shared/common/operations/test_runner.go:62]: I have seen a few `t.Fatal()` in these tests, should these instances not be replaced with eg:
```suggestion
		require.NoError(t, err, "Did not fail when expected")
```

**Review by saolyn — APPROVED** [review]
If the `Fatal` are fine then the PR looks good to me! I really like the organization and simplifications you have done, it makes a huge difference

**Pull request merged**